src/mmw/js/src/modeling/tests.js

```diff
@@ -556,21 +693,92 @@ describe('Modeling', function() {
                     this.clock = sinon.useFakeTimers();
 
                     var model = getTestScenarioModel();
-
                     this.server.respondWith('POST', '/api/modeling/start/tr55/',
-                        [200, { 'Content-Type': 'application/json' }, mocks.polling.getTR55Started]);
+                                            [200, { 'Content-Type': 'application/json' }, mocks.polling.getTR55Started]);
                     this.server.respondWith('GET', '/api/modeling/jobs/8aef636e-2079-4f87-98dc-471d090141ad/',
-                        [400, { 'Content-Type': 'application/json' }, mocks.polling.getJobFailure]);
+                                            [400, { 'Content-Type': 'application/json' }, mocks.polling.getJobFailure]);
 
                     model.addModification(new models.ModificationModel(mocks.modifications.sample1));
 
                     this.clock.tick(1000);
+                    makeAssertions();
+                    this.clock.restore();
+                });
+            });
 
-                    assert.isTrue(resultSpy.called, 'getResults should have been called');
-                    assert.isTrue(saveSpy.called, 'attemptSave should have been called');
-                    assert.isTrue(resultSpy.calledBefore(saveSpy));
+            describe('#getResults', function() {
+                beforeEach(function() {
+                    this.results = new models.ResultCollection();
+                    this.setPollingSpy = sinon.spy(this.results, 'setPolling');
+                    this.setNullResultsSpy = sinon.spy(this.results, 'setNullResults');
+                    this.scenarioModel = new models.ScenarioModel({});
+                    this.setResultsSpy = new sinon.spy(this.scenarioModel, 'setResults');
+                    this.scenarioModel.set('results', this.results);
+                    this.server = sinon.fakeServer.create();
+                    this.server.respondImmediately = true;
+                    this.startJob = '1';
+                    this.startResponse = {
+                        job: this.startJob
+                    };
+                    this.server.respondWith('POST', '/api/modeling/start/tr55/',
+                                            [ 200,
+                                              { 'Content-Type': 'application/json' },
+                                              JSON.stringify(this.startResponse) ]);
+                    this.pollingResponse = mocks.polling.getJobSuccess;
+                    this.server.respondWith('GET', '/api/modeling/jobs/1/',
+                                            [ 200,
+                                              { 'Content-Type': 'application/json' },
+                                              this.pollingResponse ]);
+                });
+
+                function getResultsAssertions(self) {
+                    assert(inputModSpy.calledTwice, 'updateInputModHash should have been called twice');
```

Comment:
This went from being called once to twice because we are now initializing the model before each test and it is called once in `initialize` and once in `getResults`, correct?